src/javascript/class-scanner.ts

```diff
@@ -594,70 +668,72 @@ export function extractPropertiesFromConstructor(
     return properties;
   }
   for (const method of astNode.body.body) {
-    if (!babel.isClassMethod(method) || method.kind !== 'constructor') {
+    if (!babel.isClassMethod(method)) {
       continue;
     }
-    const constructor = method;
-    for (const statement of constructor.body.body) {
-      if (!babel.isExpressionStatement(statement)) {
-        continue;
-      }
-      let name;
-      let astNode;
-      let defaultValue;
-      if (babel.isAssignmentExpression(statement.expression)) {
-        // statements like:
-        // /** @public The foo. */
-        // this.foo = baz;
-        name = getPropertyNameOnThisExpression(statement.expression.left);
-        astNode = statement.expression.left;
-        defaultValue = generate(statement.expression.right).code;
-      } else if (babel.isMemberExpression(statement.expression)) {
-        // statements like:
-        // /** @public The foo. */
-        // this.foo;
-        name = getPropertyNameOnThisExpression(statement.expression);
-        astNode = statement;
-      } else {
-        continue;
+
+    if (method.kind === 'constructor') {
+      const constructor = method;
+      for (const statement of constructor.body.body) {
+        if (!babel.isExpressionStatement(statement)) {
+          continue;
+        }
+        const prop = extractPropertyFromExpressionStatement(statement, document);
```

Comment:
This organization is a bit confusing for the reader, and it's a bit weird to extract a property from a generalized statement. The function only really makes sense for a constructor body.

What about renaming extractPropertiesFromConstructor to `extractPropertiesFromClass`, then putting all of the constructor handling code in an `extractPropertiesFromConstructor` function.

Reply:
yup im currently refactoring this a bit as its nasty right now :zipper_mouth_face: